app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -428,8 +428,9 @@ private void processBarcodeValuesList(List<BarcodeValues> barcodeValuesList, Str
             public void onUserChoseBarcode(BarcodeValues barcodeValues) {
                 Intent newIntent = new Intent(getApplicationContext(), LoyaltyCardEditActivity.class);
                 Bundle newBundle = new Bundle();
-                newBundle.putString(LoyaltyCardEditActivity.BUNDLE_BARCODETYPE, barcodeValues.format());
-                newBundle.putString(LoyaltyCardEditActivity.BUNDLE_CARDID, barcodeValues.content());
+                newBundle.putString(LoyaltyCard.BUNDLE_LOYALTY_CARD_BARCODE_TYPE, barcodeValues.format());
+                newBundle.putString(LoyaltyCard.BUNDLE_LOYALTY_CARD_CARD_ID, barcodeValues.content());
+                newBundle.putString(LoyaltyCard.BUNDLE_LOYALTY_CARD_BARCODE_ID, null);
```

Comment:
Perhaps this could use some wrapper? We now in multiple places translate a `BarcodeValues` instance to a bundle. Maybe `BarcodeValues` should have a `toBundle` function?

Reply:
Only one place left, so no longer relevant